components/gitpod-db/src/typeorm/team-db-impl.ts

```diff
@@ -192,6 +197,21 @@ export class TeamDBImpl implements TeamDB {
         await membershipRepo.save(membership);
     }
 
+    public async setTeamMemberSubscription(userId: string, teamId: string, subscriptionId: string): Promise<void> {
+        const teamRepo = await this.getTeamRepo();
+        const team = await teamRepo.findOne(teamId);
+        if (!team || !!team.deleted) {
+            throw new Error("A team with this ID could not be found");
+        }
+        const membershipRepo = await this.getMembershipRepo();
+        const membership = await membershipRepo.findOne({ teamId, userId, deleted: false });
+        if (!membership) {
+            throw new Error("The user is not currently a member of this team");
+        }
+        membership.subscriptionId = subscriptionId;
```

Comment:
from #9655 I can read that the subscription already knows the membership. why storing this link in both directions?

Reply:
See https://github.com/gitpod-io/gitpod/pull/9655#issuecomment-1120769840

Reply:
(Let's use this as a thread in here.)

> For example, when you leave a team, the membership is automatically deleted (leaving no trace), but the subscription is merely cancelled at end of term. This leaves a trace (e.g. for users wondering why they have unlimited hours without a paid subscription, we could see that the still-active subscription came from a team membership that has since been deleted).

So, what is the `DBTeamMembership.subscriptionId` need for then? I couldn't spot any usage.

Reply:
Sure! So, while this direction of the link is indeed not used in the current code, it was requested in the RFC that this link be materialized in both directions:

> - We add a 0..1 reference from `d_b_team_member[ship]`  to `d_b_subscription` and the opposite reference in analogy to the reference to `d_b_team_subscription_slot`. 

[Source](https://www.notion.so/gitpod/Aligning-Team-Plans-and-Teams-9073db94359549918ab5c85cc364de46#e207bc6c7bf24b578fae8f7b9bd9d5d8) (internal)

And, this made sense to me at the time, because the link between `d_b_team_subscription_slot` and `d_b_subscription` also went both ways.

As mentioned above, this slight duplication has helped/saved us a few times, for example in customer requests, or to investigate recover from bugs.

Are you proposing to drop the reference from `d_b_team_membership` to `d_b_subscription` now?